engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -34,26 +36,59 @@ class PeriodicProcessService(delegateProcessManager: ProcessManager,
     def emptyCallback: RepositoryAction[Callback] = result.map(_ => () => Future.successful(()))
   }
 
-  def schedule(periodicProperty: PeriodicProperty,
+  def schedule(schedule: BasePeriodicProperty,
                processVersion: ProcessVersion,
                processJson: String): Future[Unit] = {
-    periodicProperty.nextRunAt(clock) match {
-      case Right(Some(runAt)) =>
-        logger.info("Scheduling periodic process: {} on {}", processVersion, runAt)
+    findInitialScheduleDates(schedule) match {
+      case Right(scheduleDates) =>
```

Comment:
what about the case when all cron expressions have no match in future? do we want to allow to schedule such a process?

Reply:
:+1: I forgot to keep previous behaviour... I added test